aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -420,4 +427,40 @@ protected void finalize() throws Throwable {
       }
     }
   }
+
+  @Override
+  public boolean recoverFile(String path) {
+    S3URI location = new S3URI(path, s3FileIOProperties.bucketToAccessPointMapping());
+    ListObjectVersionsIterable response =
+        client()
+            .listObjectVersionsPaginator(
+                ListObjectVersionsRequest.builder()
```

Comment:
nit: we're using a mixture of patterns (builder/request object).   Could we just switch this to:

```java
builder -> builder.bucket(location.bucket()).prefix(location.key())
```

like we do below?